src/main/java/io/netty/handler/codec/compression/BrotliDecoder.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright 2013 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.handler.codec.compression;
+
+import com.nixxcode.jvmbrotli.common.BrotliLoader;
+import com.nixxcode.jvmbrotli.dec.BrotliInputStream;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.util.Arrays;
+import java.util.List;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.ByteBufInputStream;
+import io.netty.buffer.ByteBufOutputStream;
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.handler.codec.ByteToMessageDecoder;
+import io.netty.util.internal.logging.InternalLogger;
+import io.netty.util.internal.logging.InternalLoggerFactory;
+
+/**
+ * Decompress a {@link ByteBuf} using the inflate algorithm.
+ */
+public class BrotliDecoder extends ByteToMessageDecoder {
+
+  private static final InternalLogger log =
+      InternalLoggerFactory.getInstance(BrotliDecoder.class);
+
+  static {
+    BrotliLoader.isBrotliAvailable();
+  }
+
+  /*
+  For how this value is derived, please see: `BROTLI_MAX_NUMBER_OF_BLOCK_TYPES` in these docs:
+     - https://github.com/google/brotli/blob/master/c/common/constants.h
+     - https://tools.ietf.org/html/draft-vandevenne-shared-brotli-format-01
+   */
+  private static int BROTLI_MAX_NUMBER_OF_BLOCK_TYPES = 256;
+
+  public BrotliDecoder() {
+  }
+
+  public static byte[] decompress(byte[] compressedArray) throws IOException {
```

Comment:
who is the consumer of this public method? so far i see only tests. if so, pls remove it.